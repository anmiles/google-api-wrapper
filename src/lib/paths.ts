import fs from 'fs';
import path from 'path';
import paths from './paths';

export { ensureDir, ensureFile, getProfilesFile, getSecretsFile, getCredentialsFile };
export default { ensureDir, ensureFile, getProfilesFile, getSecretsFile, getCredentialsFile };

const dirPaths = {
	input   : 'input',
	secrets : 'secrets',
};

function ensureDir(dirPath: string) {
	if (!fs.existsSync(dirPath)) {
		fs.mkdirSync(dirPath, { recursive : true });
	}
	return dirPath;
}

function ensureFile(filePath: string) {
	paths.ensureDir(path.dirname(filePath));

	if (!fs.existsSync(filePath)) {
		fs.writeFileSync(filePath, '');
	}
	return filePath;
}

function getProfilesFile() {
	return path.join(dirPaths.input, 'profiles.json');
}

function getSecretsFile(profile: string) {
	return path.join(dirPaths.secrets, `${profile}.json`);
}

function getCredentialsFile(profile: string) {
	return path.join(dirPaths.secrets, `${profile}.credentials.json`);
}
