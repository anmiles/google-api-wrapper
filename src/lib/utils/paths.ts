import path from 'path';

import type { templates } from '../renderer';

const dirPaths = {
	input    : 'input',
	secrets  : 'secrets',
	// TODO: Remove this hack after moving to React
	templates: path.relative(process.cwd(), path.join(__dirname, '../../templates')),
};

export function getProfilesFile(): string {
	return path.join(dirPaths.input, 'profiles.json');
}

export function getScopesFile(): string {
	return 'scopes.json';
}

export function getSecretsFile(profile: string): string {
	return path.join(dirPaths.secrets, `${profile}.json`);
}

export function getCredentialsFile(profile: string): string {
	return path.join(dirPaths.secrets, `${profile}.credentials.json`);
}

export function getTemplateFile(templateName: keyof typeof templates): string {
	return path.join(dirPaths.templates, `${templateName}.html`);
}
