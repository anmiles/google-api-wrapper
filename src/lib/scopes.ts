import fs from 'fs';

import { getScopesFile } from './utils/paths';

export function getScopes(): string[] {
	const scopesFile = getScopesFile();
	const scopes     = fs.getJSON<string[]>(scopesFile, () => {
		throw new Error(getScopesError(scopesFile));
	});
	return scopes;
}

function getScopesError(scopesFile: string): string {
	return [
		`File ${scopesFile} not found!`,
		`This application had to have pre-defined file ${scopesFile} that will declare needed scopes`,
	].join('\n');
}
