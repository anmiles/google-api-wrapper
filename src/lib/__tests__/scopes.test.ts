import '@anmiles/prototypes';
import mockFs from 'mock-fs';

import { getScopes } from '../scopes';
import { getScopesFile } from '../utils/paths';

const scopesFile = getScopesFile();

const scopes = [
	'https://www.googleapis.com/auth/calendar.calendars.readonly',
	'https://www.googleapis.com/auth/calendar.events',
];

beforeEach(() => {
	mockFs({
		[scopesFile]: JSON.stringify(scopes),
	});
});

afterAll(() => {
	mockFs.restore();
});

describe('src/lib/scopes', () => {
	describe('getScopes', () => {

		it('should return scopes', () => {
			const result = getScopes();

			expect(result).toEqual(scopes);
		});

		it('should throw if scopes file does not exist', () => {
			mockFs({});

			const func = (): string[] => getScopes();

			expect(func).toThrowErrorMatchingSnapshot();
		});
	});
});
