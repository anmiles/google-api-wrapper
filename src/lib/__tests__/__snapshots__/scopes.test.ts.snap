// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/lib/scopes getScopes should throw if scopes file does not exist 1`] = `
"File scopes.json not found!
This application had to have pre-defined file scopes.json that will declare needed scopes"
`;
